File: SW_Expert_Academy/java/D3_1206_View.java
import java.io.*;
import java.util.*;

class Solution {
    static int n;
    static List<Integer> nums;

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        for (int i = 1; i <= 10; i++) {
            initialize();
            input();

            bw.write("#" + i + " ");
            solution();
            bw.write("\n");
        }

        bw.flush();
        bw.close();
    }

    private static void initialize() {
        nums = new ArrayList<>();
    }

    private static void input() throws IOException {
        n = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());

        for (int i = 0; i < n; i++) {
            int a = Integer.parseInt(st.nextToken());

            nums.add(a);
        }
    }

    private static void solution() throws IOException {
        int answer = 0;

        for (int i = 2; i < nums.size() - 2; i++) {
            if (nums.get(i - 2) < nums.get(i) && nums.get(i - 1) < nums.get(i) && nums.get(i) > nums.get(i + 1) && nums.get(i) > nums.get(i + 2)) {
                int maxNeighbor = Math.max(Math.max(nums.get(i - 2), nums.get(i - 1)), Math.max(nums.get(i + 1), nums.get(i + 2)));
                answer += (nums.get(i) - maxNeighbor);
            }
        }

        bw.write(String.valueOf(answer));
    }
}

File: SW_Expert_Academy/cpp/D3_1206_View.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int n;
vector<int> nums;

void initialize() {
	nums.clear();
}

void input() {
	cin >> n;

	for (int i = 0; i < n; i++) {
		int a;
		cin >> a;

		nums.push_back(a);
	}
}

void solution() {
	int answer = 0;

	for (int i = 2; i < nums.size() - 2; i++) {
		if (nums[i - 2] < nums[i] && nums[i - 1] < nums[i] && nums[i] > nums[i + 1] && nums[i] > nums[i + 2]) {
			int maxNeighbor = max(max(nums[i - 2], nums[i - 1]), max(nums[i + 1], nums[i + 2]));
			answer += (nums[i] - maxNeighbor);
		}
	}

	cout << answer;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	for (int i = 1; i <= 10; i++) {
		initialize();
		input();

		cout << "#" << i << " ";
		solution();
		cout << "\n";
	}

	return 0;
}